src/test/scala/mesosphere/marathon/state/ContainerTest.scala

```diff
@@ -152,7 +151,7 @@ class ContainerTest extends MarathonSpec with Matchers {
     val proto = ContainerSerializer.toProto(f.container)
     assert(mesos.ContainerInfo.Type.DOCKER == proto.getType)
     assert("group/image" == proto.getDocker.getImage)
-    assert(f.container.volumes == proto.getVolumesList.asScala.map(Volume(_)))
+    assert(f.container.volumes == proto.getVolumesList.asScala.map(Volume.fromProto(_)))
```

Comment:
I am not sure that I understand your comment completely. Anyway, having it separate and/or in state.serialization or something like that sounds great (it has nothing to do with the api).